### PR TITLE
Merge release/3.0 to dev branch

### PR DESCRIPTION
**Pull request by cratz**
Done in a separate branch because of conflicts.

### CODE REVIEW

File: spec/controllers/api/v2/planning_element_types_controller_spec.rb [hunk 1]
```diff
@@ -32,6 +32,7 @@
   let (:current_user) { FactoryGirl.create(:admin) }
 
   before do
+    @controller.stub(:require_login)
```
Comment (hschink) [inline spec/controllers/api/v2/planning_element_types_controller_spec.rb]: This should become `allow` too.

File: spec/controllers/types_controller_spec.rb [hunk 2]
```diff
@@ -38,207 +38,216 @@
...
-    render_views
-    let(:type) { FactoryGirl.create(:type, :name => 'My type', :is_milestone => true, :projects => [project]) }
+  context "with an authorized account" do
+    let(:current_user) { FactoryGirl.create(:user) }
 
     before do
-      get 'edit', :id => type.id
+      @controller.stub(:require_login)
+      @controller.stub(:check_if_login_required)
+      @controller.stub(:require_admin)
```
Comment (hschink) [inline spec/controllers/types_controller_spec.rb]: These should use `allow` too.

File: spec/controllers/work_packages_controller_spec.rb [hunk 3]
```diff
@@ -526,6 +526,8 @@ def self.requires_export_permission(&block)
                                           .at_most(:twice)
                                           .with(:project => stub_work_package.project)
                                           .and_return(wp_params)
+
+        current_user.should_receive(:allowed_to?).with(:edit_work_packages, stub_work_package.project).and_return(true);
```
Comment (hschink) [inline spec/controllers/work_packages_controller_spec.rb]: This, then, becomes `expect(current_user).to receive(:allowed_to?)...`.

**Comment by cratz** [comment]
Thanks for the fixes :D

**Pull request merged**